javascript/ql/lib/semmle/javascript/frameworks/Hapi.qll

```diff
@@ -9,10 +9,21 @@ module Hapi {
   /**
    * An expression that creates a new Hapi server.
    */
-  class ServerDefinition extends Http::Servers::StandardServerDefinition, DataFlow::NewNode {
+  class ServerDefinition extends Http::Servers::StandardServerDefinition, DataFlow::Node {
     ServerDefinition() {
       // `server = new Hapi.Server()`
       this = DataFlow::moduleMember("hapi", "Server").getAnInstantiation()
+      or
+      // server = Glue.compose(manifest, composeOptions)
+      this = DataFlow::moduleMember("@hapi/glue", "compose").getAnInvocation()
+      or
+      // server inside a plugin
+      // TODO match `function (server, options)`
+      exists(Function f |
+        this.(DataFlow::ParameterNode).getParameter() = f.getParameter(0) and
+        f.getParameter(0).getName() = "server" and
+        f.getParameter(1).getName() = "options"
+      )
```

Comment:
How is this `function (server, options)` used?  
I suppose there is some call to Hapi that registers the plugin?   
I think that registration call should be the `ServerDefinition`.

Reply:
have you tried using the registration as definition?